examples/qbf/Coloring.hs

```diff
@@ -0,0 +1,112 @@
+module Main where
+
+import Prelude hiding (not, (&&), and, or)
+
+import Ersatz
+import Ersatz.Relation
+import Ersatz.Counting
+
+import qualified Data.Array as A
+import Data.Array (Array, Ix)
+
+import Control.Monad.State.Lazy (StateT)
+
+import Data.List (tails)
+
+import qualified Data.ByteString.Lazy as B
+import qualified Data.ByteString.Lazy.Char8 as C
+import Data.Functor.Identity
+
+
+-- | Grötzsch graph: not 3-colorable and K3-free (11 nodes)
+main :: IO ()
+main = do
+  let n = 11 -- number of nodes
+      c = 3  -- not 3-colorable
+      k = 3  -- K3-free
+  formulaSize $ problem n c k
+  solve $ problem n c k
+
+-- | not 4-colorable and K3-free (21 or 22 nodes?)
+variant_a :: IO ()
+variant_a = do
+  let n = 22; c = 4; k = 3
+  formulaSize $ problem n c k
+  solve $ problem n c k
+
+-- | not 4-colorable and K4-free (11 nodes)
+variant_b :: IO ()
+variant_b = do
+  let n = 11; c = 4; k = 4
+  formulaSize $ problem n c k
+  solve $ problem n c k

```

Comment:
Neither of these `variant`s are actually used at runtime. Should they be?

Reply:
I added a small parser instead.